src/lib.rs

```diff
@@ -1215,6 +1217,22 @@ enum SignAlgo {
 	Rsa(),
 }
 
+impl Hash for SignAlgo {
+	fn hash<H: Hasher>(&self, state: &mut H) {
+		match self {
+			SignAlgo::EcDsa(_) => {
+				"SignAlgo::EcDsa".hash(state);
+			}
+			SignAlgo::EdDsa(_) => {
+				"SignAlgo::EdDsa".hash(state);
```

Comment:
This strips away all the useful data inside the SignAlgo, especially which *kind* of EdDsa algo it is. Instead I think you should convert the interior pointer to a static into an integer and hash that value. Alternatively one can extend the enum with another field and use that field for hashing.

Reply:
For my purpose this is sufficient, but the only way to retain `EcdsaSigningAlgorithm` or `EdDSAParameters` is via the `Debug` trait. See https://github.com/zurborg/rcgen/commit/14232c8d24f71cc2328f6879fc5e3583e763073d

Reply:
It might be sufficient for your purpose, but it would be quite unexpected for people if the hash were the same for two distinct signature algorithms.